tests/migration/namespace.go

```diff
@@ -0,0 +1,406 @@
+)
+
+var _ = Describe(SIG("Live Migration across namespaces", Serial, decorators.RequiresDecentralizedLiveMigration, func() {
+	var (
+		virtClient         kubecli.KubevirtClient
+		migrationID        string
+		connectionURL      string
+		err                error
+		featureGateEnabled bool
+	)
+
+	BeforeEach(func() {
+		featureGateEnabled = checks.HasFeature(featuregate.DecentralizedLiveMigration)
+		if !featureGateEnabled {
+			kvconfig.EnableFeatureGate(featuregate.DecentralizedLiveMigration)
+		}
+		if !libstorage.HasCDI() {
+			Fail("Fail DataVolume tests when CDI is not present")
+		}
+		virtClient = kubevirt.Client()
+		migrationID = fmt.Sprintf("mig-%s", rand.String(5))
+		connectionURL, err = getKubevirtSynchronizationSyncAddress(virtClient)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if !featureGateEnabled {
+			kvconfig.DisableFeatureGate(featuregate.DecentralizedLiveMigration)
+		}
+	})
+
+	createAndStartVMFromVMISpec := func(vmi *virtv1.VirtualMachineInstance) *virtv1.VirtualMachine {
+		vm := libvmi.NewVirtualMachine(vmi)
+		vm, err := virtClient.VirtualMachine(vmi.Namespace).Create(context.Background(), vm, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Starting the VM")
+		vm = libvmops.StartVirtualMachine(vm)
+		vmi = libwait.WaitForVMIPhase(vmi, []v1.VirtualMachineInstancePhase{v1.Running})
+		_, err = libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+		Expect(err).NotTo(HaveOccurred())
+
+		return vm
+	}
+
+	createReceiverVMFromVMISpec := func(vmi *virtv1.VirtualMachineInstance) *virtv1.VirtualMachine {
+		vm := libvmi.NewVirtualMachine(vmi,
+			libvmi.WithRunStrategy(virtv1.RunStrategyWaitAsReceiver),
+		)
+		By(fmt.Sprintf("creating VM %s/%s", vmi.Namespace, vmi.Name))
+		vm, err := virtClient.VirtualMachine(vmi.Namespace).Create(context.Background(), vm, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Checking the VMI exists in receiving phase")
+		Eventually(func() virtv1.VirtualMachineInstancePhase {
+			receiver, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+			if err != nil {
+				return ""
+			}
+			return receiver.Status.Phase
+		}, 30*time.Second, 1*time.Second).Should(Equal(virtv1.WaitingForSync))
+
+		return vm
+	}
+
+	deleteMigration := func(migration *virtv1.VirtualMachineInstanceMigration) error {
+		err := virtClient.VirtualMachineInstanceMigration(migration.Namespace).Delete(context.Background(), migration.Name, metav1.DeleteOptions{})
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+		// Verify migration is gone
+		Eventually(func() *virtv1.VirtualMachineInstanceMigration {
+			migration, err := virtClient.VirtualMachineInstanceMigration(migration.Namespace).Get(context.Background(), migration.Name, metav1.GetOptions{})
+			if k8serrors.IsNotFound(err) {
+				return nil
+			}
+			return migration
+		}, 30*time.Second, 1*time.Second).Should(BeNil())
```

Comment:
Maybe this can be done with something like that?
```suggestion
		Eventually(ThisMigration(migration), 30*time.Second, 1*time.Second).Should(BeNil())
```